src/metabase/api/field.clj

```diff
@@ -147,6 +171,7 @@
                                       :present #{:caveats :description :fk_target_field_id :points_of_interest :semantic_type :visibility_type
                                                  :coercion_strategy :effective_type :has_field_values :nfc_path :json_unfolding}
                                       :non-nil #{:display_name :settings}))))
+    (update-nested-fields-on-json-unfolding-change! field json_unfolding)
```

Comment:
wrap this with a `(when (some? json_unfolding)` ?

Reply:
Updated and tested https://github.com/metabase/metabase/commit/dee4122458c0d60a1cffefd3f6c4d21b2e1f0561